app/scripts/components/analysis/utils.ts

```diff
@@ -0,0 +1,58 @@
+import { userTzDate2utcString } from '$utils/date';
+import { endOfDay, startOfDay } from 'date-fns';
+import { Feature, MultiPolygon, Polygon } from 'geojson';
+
+/**
+ * Creates the appropriate filter object to send to STAC.
+ *
+ * @param {Date} start Start date to request
+ * @param {Date} end End date to request
+ * @param {string} collection STAC collection to request
+ * @returns Object
+ */
+export function getFilterPayload(
+  start: Date,
+  end: Date,
+  aoiFeature: Feature<Polygon>,
+  collections: string[]
+) {
+  const filterPayload = {
+    op: 'and',
+    args: [
+      {
+        op: 't_intersects',
+        args: [
+          { property: 'datetime' },
+          {
+            interval: [
+              userTzDate2utcString(startOfDay(start)),
+              userTzDate2utcString(endOfDay(end))
+            ]
+          }
+        ]
+      },
+      {
+        op: 's_intersects',
+        args: [{ property: 'geometry' }, aoiFeature.geometry]
+      },
+      {
+        op: 'in',
+        args: [{ property: 'collection' }, collections]
+      }
+    ]
+  };
+  return filterPayload;
+}
+
+export function multiPolygonToPolygon(
```

Comment:
This function's name seems a bit misleading. You're effectively removing part of the multipolygon to return a polygon, not doing a direct conversion.

Reply:
I agree that's not 100% accurate, but honestly I'm struggling to find a decent name for that 😄 (`firstMultiPolygonPolygonToPolygon` ? `trimMultiPolygonToPolygon` ?)
I'd argue that there's no ambiguity since there's no way to _convert_ a Polygon to a MultiPolygon, so one could infer that it's taking the first MultiPolygon's polygon?

Reply:
fair point!

Reply:
We don't need to do that, but I imagine that this function could exist with a second parameter to indicate the index of the multipolygon's polygon to keep. :D

Reply:
Yes, but [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it) 😁